baseplate/core.py

```diff
@@ -105,7 +109,15 @@ def from_upstream(cls, trace_id, parent_id, span_id):
         if parent_id is None or not 0 <= parent_id < 2**64:
             raise ValueError("invalid parent_id")
 
-        return cls(trace_id, parent_id, span_id)
+        if sampled is not None and (sampled not in ["0", "1"]):
+            raise ValueError("invalid sampled header")
+
+        if flags is not None:
+            flags = int(flags)
```

Comment:
also good to require the caller to make this an int. worth doing an enum for values and checking against that? or shall we let things fly by that we don't understand for better compatibility?